MekHQ/src/mekhq/campaign/universe/IUnitGenerator.java

```diff
@@ -12,80 +13,79 @@
  *
  * @author Neoancient
- *
  */
 public interface IUnitGenerator {
-
-    /*
+    /**
      * For convenience in generating traditional ground + vtol units.
      */
-    final static EnumSet<EntityMovementMode> MIXED_TANK_VTOL = EnumSet.of(EntityMovementMode.TRACKED,
+    EnumSet<EntityMovementMode> MIXED_TANK_VTOL = EnumSet.of(EntityMovementMode.TRACKED,
             EntityMovementMode.WHEELED, EntityMovementMode.HOVER, EntityMovementMode.WIGE,
             EntityMovementMode.VTOL);
 
     /**
      * For convenience in generating infantry units.
      */
-    final static EnumSet<EntityMovementMode> ALL_INFANTRY_MODES = EnumSet.of(EntityMovementMode.INF_JUMP,
+    EnumSet<EntityMovementMode> ALL_INFANTRY_MODES = EnumSet.of(EntityMovementMode.INF_JUMP,
             EntityMovementMode.INF_LEG, EntityMovementMode.INF_MOTORIZED, EntityMovementMode.INF_UMU,
             EntityMovementMode.TRACKED, EntityMovementMode.WHEELED, EntityMovementMode.HOVER);
-    
-    final static EnumSet<EntityMovementMode> ALL_BATTLE_ARMOR_MODES = EnumSet.of(EntityMovementMode.INF_JUMP,
+
+    EnumSet<EntityMovementMode> ALL_BATTLE_ARMOR_MODES = EnumSet.of(EntityMovementMode.INF_JUMP,
             EntityMovementMode.INF_LEG, EntityMovementMode.INF_UMU, EntityMovementMode.VTOL);
 
     /**
      * For convenience in generating infantry units, the minimum tonnage of a foot infantry platoon.
      */
-    final static double FOOT_PLATOON_INFANTRY_WEIGHT = 3.0;
-    
+    double FOOT_PLATOON_INFANTRY_WEIGHT = 3.0;
+
     /**
      * For convenience in generating battle armor, minimum tonnage of a battle armor squad.
      */
-    final static double BATTLE_ARMOR_MIN_WEIGHT = 4.0;
-    
+    double BATTLE_ARMOR_MIN_WEIGHT = 4.0;
+
     /**
-     * For convenience in generating battle armor/infantry, when the tonnage does not matter 
-     * (a dedicated dropship bay, battle armor riding on a mech, etc)
+     * For convenience in generating battle armor/infantry, when the tonnage does not matter
+     * (a dedicated DropShip bay, battle armor riding on a 'Mech, etc)
      */
-    final static double NO_WEIGHT_LIMIT = -1.0;
+    double NO_WEIGHT_LIMIT = -1.0;
 
     /**
      * Convenience function to let us know whether a unit type supports weight class selection.
      * @param unitType The unit type to check.
      * @return Whether or not the unit type supports weight class selection.
      */
-    static boolean unitTypeSupportsWeightClass(int unitType) {
-        return unitType == UnitType.AERO || unitType == UnitType.MEK || unitType == UnitType.TANK;
+    static boolean unitTypeSupportsWeightClass(final int unitType) {
+        return (unitType == UnitType.AERO) || (unitType == UnitType.MEK) || (unitType == UnitType.TANK);
     }
 
     /**
-     *
      * @param unitType UnitType constant
      * @return true if the generator supports the unit type
      */
-    boolean isSupportedUnitType(int unitType);
+    boolean isSupportedUnitType(final int unitType);
```

Comment:
I don't believe `final` on a method in an interface has any bearing on the actual implementation.

I think they're fine on the `default` implementations, as those would be respected, but the rest are free to be ignored by implementations.

Reply:
I've added them with carry through, so you can just look at the interface and know how they are in all implementations of it. Hence why no predicates are final, because the value is mutable in one of the methods.

However, I will likely remove the interface in the nearish future as part of moving RATs out of AtB and into their own options category, as it would be better implemented in the abstract class. This is a stepping stone towards it that fixes an annoying bug.